### PR TITLE
[16.0][IMP] base_report_to_printer: Add neutralization queries

### PR DESCRIPTION
**Pull request by micheledic**
### Before this PR
There is no way to neutralize printers and printer server using neutralization function

### After this PR
Since 16.0 of codebase there is a new function "Neutralization" that runs all neutralization queries that basically deactivate ir_cron, ir_mail_server and other production stuff. With this PR i ADD neutralization queries  that deactivates printing server and printer during neutralization .

### CODE REVIEW

**Comment by micheledic** [comment]
The first check is failed but not caused by my change

**Comment by micheledic** [comment]
@nilshamerlinck @pedrobaeza @MiquelRForgeFlow

**Comment by MiquelRForgeFlow** [comment]
> The first check is failed but not caused by my change

Yes, it is.

Selection_1348

**Comment by micheledic** [comment]
> > The first check is failed but not caused by my change
> 
> Yes, it is.
> 
> Selection_1348

hello,
the neutralization queries are not declared in any files so it is correct that it is not used from manifest. That error can be ignored

Since odoo 16.0 , the neutralization queries are declared inside data path and the neutralization function scans every "data" folder of the modules to check the existence of .sql .

File: base_report_to_printer/data/neutralize.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+update printing_printer set active=false;
```
Comment (MiquelRForgeFlow) [inline base_report_to_printer/data/neutralize.sql:2]:
```suggestion
# pylint: disable=file-not-used
update printing_printer set active=false;
```

Comment (micheledic) [reply base_report_to_printer/data/neutralize.sql:2]: file changed!

Comment (micheledic) [reply base_report_to_printer/data/neutralize.sql:2]: @MiquelRForgeFlow same error...

Comment (MiquelRForgeFlow) [reply base_report_to_printer/data/neutralize.sql:2]: Maybe the pylint exception has to be put in another place... @moylop260 do you know?

Comment (moylop260) [reply base_report_to_printer/data/neutralize.sql:2]: Hi everybody

The pylint-odoo version used is too old:
 - https://github.com/OCA/report-print-send/blob/af5df33b010dc671ca33a6b4d4fc0867a7718ae2/.pre-commit-config.yaml#L128

The latest one is https://pypi.org/project/pylint-odoo/8.0.19

@yajo updated the template in order to update the pylint configuration file too

Regards!

Comment (micheledic) [reply base_report_to_printer/data/neutralize.sql:2]: > Hi everybody
> 
> The pylint-odoo version used is too old:
>  - https://github.com/OCA/report-print-send/blob/af5df33b010dc671ca33a6b4d4fc0867a7718ae2/.pre-commit-config.yaml#L128
> 
> The latest one is https://pypi.org/project/pylint-odoo/8.0.19
> 
> @yajo updated the template in order to update the pylint configuration file too
> 
> Regards!

Hello, now what should i do? The yaml in my PR Is old. How can i change the head of the PR?

Comment (yajo) [reply base_report_to_printer/data/neutralize.sql:2]: First, we need to merge https://github.com/OCA/oca-addons-repo-template/pull/198. Then do a template release and update the template here. Then, you can rebase the PR and all will get green.

**Comment by MiquelRForgeFlow** [comment]
@micheledic could you rebase?

**Comment by MiquelRForgeFlow** [comment]
I mean, maybe we should update the copier template.

**Comment by micheledic** [comment]
> @micheledic could you rebase?

hello, rebased!

**Comment by micheledic** [comment]
@MiquelRForgeFlow  error in pre-commit

**Comment by MiquelRForgeFlow** [comment]
> I mean, maybe we should update the copier template.

Then do `copier -f update`

File: base_report_to_printer/data/neutralize.sql (hunk 1 above)
Comment (MiquelRForgeFlow) [reply base_report_to_printer/data/neutralize.sql:2]: @yajo the error persists...

**Comment by micheledic** [comment]
> > I mean, maybe we should update the copier template.
> 
> Then do `copier -f update`

Sorry the files keeps remaining untouched also after copier -f update...

**Comment by micheledic** [comment]
Every time i run the update, this appears and the files remains untouched
```
odoo@odoo-VirtualBox:~/report-print-send$ copier update -f --UNSAFE
Updating to template version 1.14.2

Copying from template version 1.14.2
 identical  .
 identical  .copier-answers.yml
 identical  .prettierrc.yml
 identical  .eslintrc.yml
  conflict  .pre-commit-config.yaml
 overwrite  .pre-commit-config.yaml
 identical  .github
 identical  .github/workflows
 identical  .github/workflows/test.yml
 identical  .github/workflows/stale.yml
 identical  .github/workflows/pre-commit.yml
  conflict  README.md
 overwrite  README.md
    create  .t2d.yml
 identical  .editorconfig
 identical  .pylintrc
 identical  .isort.cfg
 identical  .gitignore
 identical  LICENSE
 identical  .pylintrc-mandatory
 identical  .flake8

 > Running task 1 of 1: rm -f .travis.yml .t2d.yml

```

**Comment by yajo** [comment]
It seems a bug in pylint-odoo: https://github.com/OCA/pylint-odoo/issues/464

**Comment by micheledic** [comment]
Bumped version of template with PR https://github.com/OCA/report-print-send/pull/334

than i try to rebase and check the errors

**Comment by micheledic** [comment]
Rebased (so it has the new 1.15.0 copier template https://github.com/OCA/report-print-send/pull/334 ) and all checks ok! @pedrobaeza  @MiquelRForgeFlow

**Review by MiquelRForgeFlow — APPROVED** [review]
Finally!

**Comment by pedrobaeza** [comment]
Just a minor nitpicking: this is not `[ADD]`, but `[IMP]`, as you are adding a feature inside an existing module, not adding a module.

**Comment by micheledic** [comment]
> Just a minor nitpicking: this is not `[ADD]`, but `[IMP]`, as you are adding a feature inside an existing module, not adding a module.

should i change the commit message ?

**Comment by pedrobaeza** [comment]
It would be good

**Comment by micheledic** [comment]
> It would be good

done! sorry for before i was migrating printer_zpl2 https://github.com/OCA/report-print-send/pull/335 and i have pushed to the wrong branch

**Comment by pedrobaeza** [comment]
Thanks

/ocabot merge patch

**Pull request merged**